### PR TITLE
hw4 (lists and RecyclerView)

### PR DESCRIPTION
**Pull request by alexey-chaichuk**
Screenshot_1607280453
Screenshot_1607280458

### CODE REVIEW

File: app/src/main/java/ru/chaichuk/hwapp/ActorsListAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+import androidx.core.content.res.ResourcesCompat
+import androidx.recyclerview.widget.RecyclerView
+import ru.chaichuk.hwapp.data.models.Actor
+
+class ActorsListAdapter() : RecyclerView.Adapter<ActorsViewHolder>() {
+    private var actors = listOf<Actor>()
+
+    override fun getItemViewType(position: Int): Int {
+        return when (actors.size) {
+            0 -> VIEW_TYPE_EMPTY
```
Comment (Axrorxoja) [inline app/src/main/java/ru/chaichuk/hwapp/ActorsListAdapter.kt]: why need VIEW_TYPE_EMPTY view type ?

File: app/src/main/java/ru/chaichuk/hwapp/ActorsListAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+    override fun getItemCount(): Int = actors.size
+
+    fun bindActors(newActors : List<Actor>) {
+        actors = newActors
+        notifyDataSetChanged()
+    }
+}
+
+abstract class ActorsViewHolder(itemView: View): RecyclerView.ViewHolder(itemView)
```
Comment (Axrorxoja) [inline app/src/main/java/ru/chaichuk/hwapp/ActorsListAdapter.kt]: why need like so parent class 
which doing nothing

File: app/src/main/java/ru/chaichuk/hwapp/MoviesLIstAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+
+class MoviesListAdapter(private val clickListener: OnRecyclerItemClicked) :
+    RecyclerView.Adapter<MoviesListViewHolder>() {
+
+    private var movies = listOf<Movie>()
+
+    override fun getItemViewType(position: Int): Int {
+        return when (movies.size) {
+            0 -> VIEW_TYPE_EMPTY
```
Comment (Axrorxoja) [inline app/src/main/java/ru/chaichuk/hwapp/MoviesLIstAdapter.kt]: here also the same situation 
why need not used viewType ?

File: app/src/main/java/ru/chaichuk/hwapp/MoviesLIstAdapter.kt [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+    override fun getItemCount(): Int = movies.size
+
+    fun bindMovies(newMovies: List<Movie>) {
+        movies = newMovies
+        notifyDataSetChanged()
+    }
+
+}
+
+abstract class MoviesListViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView)
```
Comment (Axrorxoja) [inline app/src/main/java/ru/chaichuk/hwapp/MoviesLIstAdapter.kt]: the same situation which not needed parent class

File: app/src/main/java/ru/chaichuk/hwapp/fragments/MoviesDetailsFragment.kt [hunk 5]
```diff
@@ -19,8 +24,9 @@ class MoviesDetailsFragment : Fragment(R.layout.fragment_movies_details) {
     }
 
     override fun onDetach() {
-        super.onDetach()
         listener = null
+        rv_actors = null
```
Comment (Axrorxoja) [inline app/src/main/java/ru/chaichuk/hwapp/fragments/MoviesDetailsFragment.kt]: rv_actors should be clear onDestroyView

File: app/src/main/java/ru/chaichuk/hwapp/fragments/MoviesListFragment.kt [hunk 6]
```diff
@@ -19,15 +26,44 @@ class MoviesListFragment : Fragment(R.layout.fragment_movies_list) {
     }
 
     override fun onDetach() {
-        super.onDetach()
+        rv_movies_list = null
```
Comment (Axrorxoja) [inline app/src/main/java/ru/chaichuk/hwapp/fragments/MoviesListFragment.kt]: same situation

File: app/src/main/res/layout/view_holder_movie_empty.xml [hunk 7]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (Axrorxoja) [inline app/src/main/res/layout/view_holder_movie_empty.xml]: why we need this layout

File: app/src/main/java/ru/chaichuk/hwapp/ActorsListAdapter.kt (hunk 1 above)
Comment (alexey-chaichuk) [reply app/src/main/java/ru/chaichuk/hwapp/ActorsListAdapter.kt]: copied it from workshop code to understand why I could need it in my code.
did not find any reason, but leaved it to think about it later

Comment (Axrorxoja) [reply app/src/main/java/ru/chaichuk/hwapp/ActorsListAdapter.kt]: let's remove unused codes from PR
it's not good habbit

Comment (alexey-chaichuk) [reply app/src/main/java/ru/chaichuk/hwapp/ActorsListAdapter.kt]: removed all that empty view holders

File: app/src/main/java/ru/chaichuk/hwapp/ActorsListAdapter.kt (hunk 2 above)
Comment (alexey-chaichuk) [reply app/src/main/java/ru/chaichuk/hwapp/ActorsListAdapter.kt]: removed

File: app/src/main/java/ru/chaichuk/hwapp/MoviesLIstAdapter.kt (hunk 3 above)
Comment (alexey-chaichuk) [reply app/src/main/java/ru/chaichuk/hwapp/MoviesLIstAdapter.kt]: removed

File: app/src/main/java/ru/chaichuk/hwapp/fragments/MoviesDetailsFragment.kt (hunk 5 above)
Comment (alexey-chaichuk) [reply app/src/main/java/ru/chaichuk/hwapp/fragments/MoviesDetailsFragment.kt]: it was my negligence - fixed

File: app/src/main/java/ru/chaichuk/hwapp/fragments/MoviesListFragment.kt (hunk 6 above)
Comment (alexey-chaichuk) [reply app/src/main/java/ru/chaichuk/hwapp/fragments/MoviesListFragment.kt]: fixed

File: app/src/main/res/layout/view_holder_movie_empty.xml (hunk 7 above)
Comment (alexey-chaichuk) [reply app/src/main/res/layout/view_holder_movie_empty.xml]: deleted

**Review by Axrorxoja — APPROVED** [review]

**Pull request merged**